### PR TITLE
Fixes #12514

### LINKED ISSUE #12514: OpenBaseDIR BUG
As report by Emanuel Bronshtein,
https://github.com/phpmyadmin/phpmyadmin/blob/251cd5d071b194720a931cc2aa5591553fe3872b/libraries/Encoding.php#L283
    $tmpfname = tempnam('', $enc);
according to:
    https://secure.php.net/manual/en/function.tempnam.php#93256

### PR DESCRIPTION
**Pull request by osaid-r**
Before submitting pull request, please check that every commit:

- [x] Has proper Signed-Off-By
- [X] Has commit message which describes it
- [x] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [x] Any new functionality is covered by tests

Changed Encoding.php and File.php such that it takes temp directory from configfile.php

### CODE REVIEW

**Comment by ragnerok** [comment]
I have a few doubts 
1. Do I have to sign off somewhere ? Cause I haven't . Although I don't plan to not agree with the agreement
2. I do not get the error that has been pointed out by the bot !

**Comment by ibennetch** [comment]
Hello @ragnerok and thanks for your contribution.

Yes, for legal purposes we require agreement with the [DCO document](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO); the best way to do so is when committing in git with something like `git commit -s`. This automatically adds the proper signed-off-by line to the commit message.

The bot is stating that your commit adds a newline character at the end of libraries/config/ConfigFile.php which was probably done automatically by your editor. If I recall correctly, our template system does not deal well with newlines at the end of certain files, so the bot looks for this.

Finally, you seem to have added a few files that you have added some files that seem to be unnecessary (certainly .DS_Store, but probably also a few others), so it would be good if you could remove those from your commit.

I and the others might have some comment on specific parts of the code, but that should at least answer your questions so far. In order to fix the signed-off-by message and extra files, you can amend your existing commit and push (with --force) the changes back to your repository on github (git and the manuals warn about rewriting history, but since your change hasn't been merged such rewrites are currently harmless). That will update this pull request automatically with the changes.

Thanks for your enthusiasm and work.

**Comment by ragnerok** [comment]
Hi, I am relatively new to git so I don't what to do to remove the extra files using git . What do I have to do exactly ?

File: libraries/config/ConfigFile.php [hunk 1]
```diff
@@ -537,4 +537,24 @@ public function getConfigArray()
...
+    {
+        $tmp_subdir = null;
+        if (! empty($GLOBALS['cfg']['TempDir']) && @is_writable($GLOBALS['cfg']['TempDir'])) {
+        $tmp_subdir = $GLOBALS['cfg']['TempDir'];
+        } else {
+            $tmp_subdir = ini_get('upload_tmp_dir');
+            if (empty($tmp_subdir)) {
+                $tmp_subdir = sys_get_temp_dir();
+            }
+        $tmp_subdir = rtrim($tmp_subdir, DIRECTORY_SEPARATOR);
```
Comment (nijel) [inline libraries/config/ConfigFile.php]: Can you please fix indentation in this function?

**Review by nijel — CHANGES_REQUESTED** [review]
There is another copy of similar code in prefs_manage.php, can you please replace it with new function as well?

File: libraries/config/ConfigFile.php (hunk 1 above)
Comment (ragnerok) [inline libraries/config/ConfigFile.php]: yeah ok i'll work on both of these

**Comment by ragnerok** [comment]
@nijel i am not able to find the similar code in prefs_manage.php

**Comment by nijel** [comment]
It's slightly different, and starts here: https://github.com/phpmyadmin/phpmyadmin/blob/master/prefs_manage.php#L73

**Comment by ragnerok** [comment]
I kind of get how this works , correct me if I am wrong
we'll first call the function that I have added and then if it's returned value ==$GLOBALS['cfg']['TempDir']  then the code will be same from line 83-86

**Comment by nijel** [comment]
Actually it should share even more code, I've adjusted it in df66e4fa080e2aa938bc415a5be9bfe7af611301.

**Comment by nijel** [comment]
Thanks, your contributions have been merged as b8a9b0f30d3d83746967ef457cca641e014c32a5 (after rebasing and squashing the commits). Additional fix was needed to make it work: 84971eaf4f1b47a148e60a0042291e5ccb45ea3d

**Pull request closed without merging**